### ISSUE TITLE
Possibility to change event color

### ISSUE THREAD

**Issue by pulcov** [owncloud/calendar#120]
Hi,

I think that would be a nice improvement to integrate the possibility to change the color of the single event without change calendar color.

Thanks.

**Comment by georgehrke** [comment]
please describe a use-case! I don't see how this can be useful.

**Comment by phsc** [comment]
Me personally, I use it to make it clear at first sight, that I will be absent (e.g. marking holidays).
Outlook offers this possibility. But probably something for later. No killer feature in my eyes.

**Comment by pulcov** [comment]
Change color is used to highlight item like holidays, very important appointment, ...
I understand that is not a killer feature, but this is a basic feature of calendar. This fonctionnality is integrate on all calendar apps (Outlook, Google calendar, Lightning, and other).

**Comment by georgehrke** [comment]
And what's wrong with creating a dedicated calendar for holidays or very important events?

**Comment by pulcov** [comment]
No wrong, just weird especially if calendar is shared.

**Comment by phsc** [comment]
It's possible, but then more configuration is needed for shared calendars.
E.g. I have a calendar, that is synced to my mobile phone and my wife's mobile phone. Then I have to sync each dedicated calendar to each mobile phone. And everytime (ok, probably this does not happen so often) I create a new dedicated calendar, I have to touch the configuration of the mobile phones.

**Comment by georgehrke** [comment]
I don't know what mobile platform you use, but with iOS and android you only have to setup one url to sync all calendars. 
And sharing a calendar also takes only a few clicks ;)

I'm not really convinced.

cc @jancborchardt

**Comment by jancborchardt** [comment]
@pulcov @phsc can you show screenshots of how other platforms do this? Then we can consider. However, this certainly is low on the priority list as we have lots of basics to do right first.

**Comment by pulcov** [comment]
Google Calendar
![google](https://cloud.githubusercontent.com/assets/16894464/12646292/5b126a1a-c5cf-11e5-93a5-b8360b50dbc9.jpg)

**Comment by pulcov** [comment]
Microsoft Outlook 365
![outlook365](https://cloud.githubusercontent.com/assets/16894464/12646328/907a72ce-c5cf-11e5-834e-2c56b4b272db.jpg)

**Comment by phsc** [comment]
Month view:
![google_calendar_colors](https://cloud.githubusercontent.com/assets/2138442/12647400/45f69d1c-c5d5-11e5-8afc-daa8570bb06a.png)

The event on the 4th has the default color of my calendar.
The event on the 5th has an own event color (please note the green borders, which are derived from the calendar's default color).

**Comment by georgehrke** [comment]
Okay, so we might add this, but this is definitely nothing that will go into the initial release.

We should check how Google saves it. It's probably just an additional property inside the VEvent, we should be able to reuse it.

**Comment by phsc** [comment]
Hm, unfortunately it is not a VEVENT property, I think.

Event on the 4th:
```
BEGIN:VEVENT
DTSTART;VALUE=DATE:20160104
DTEND;VALUE=DATE:20160105
DTSTAMP:20160128T144532Z
UID:jmhbr19m4jn8urorb9ns831unc@google.com
CREATED:20160128T143327Z
DESCRIPTION:
LAST-MODIFIED:20160128T143327Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:test
TRANSP:TRANSPARENT
END:VEVENT
```

Event on the 5th:
```
BEGIN:VEVENT
DTSTART;VALUE=DATE:20160105
DTEND;VALUE=DATE:20160106
DTSTAMP:20160128T144532Z
UID:51k9s8vie3d4m1b7mge1oigipc@google.com
CREATED:20160128T143322Z
DESCRIPTION:
LAST-MODIFIED:20160128T143334Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:test
TRANSP:TRANSPARENT
END:VEVENT
```

The whole ICS-File (has to be renamed to .ics):
[xyz@gmail.com.txt](https://github.com/owncloud/calendar-rework/files/108328/xyz.gmail.com.txt)

**Comment by georgehrke** [comment]
Too bad :(
Anyway, thanks for taking a look :)

**Comment by pulcov** [comment]
Outlook use categories of vevent. For each categories, it is possible to set color and label.
It can be a solution to have event color while respecting the ics standard

**Comment by rschmidm** [comment]
Yes, the easiest should be to simply support event categories in the Owncloud web frontend / calendar app. The database already supports it, as I just found out:
I created an event in Outlook, assigned a category "Testproject" to it, then synced to Owncloud using cfos OutlookDAV. When I downloaded the ICS file from Owncloud, it had the Category "Testproject" in, see below. Syncing to another PC with Outlook (again with OutlookDAV), the Category was shown in Outlook.

So, my suggestion is the following:
- Add a category picker in the Calendar app to assign/change categories of events
- Have a (user defined) master list of categories with a custom color for each category
- If an event has a category, display the event in the category color instead of the calendar color in the OC Calendar app

Can this be done?

Here the ICS file:
BEGIN:VCALENDAR
PRODID:-//Microsoft Corporation//Outlook 16.0 MIMEDIR//EN
VERSION:2.0
X-MS-OLK-FORCEINSPECTOROPEN:TRUE
BEGIN:VTIMEZONE
TZID:W. Europe Standard Time
BEGIN:STANDARD
DTSTART:19701028T030000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:19700325T020000
RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VEVENT
CATEGORIES:Testproject
CLASS:PUBLIC
CREATED:20170203T174032Z
DTEND;TZID="W. Europe Standard Time":20170202T130000
DTSTAMP:20170203T174032Z
DTSTART;TZID="W. Europe Standard Time":20170202T123000
LAST-MODIFIED:20170203T174032Z
PRIORITY:5
SEQUENCE:0
SUMMARY;LANGUAGE=de-lu:Test event with category Testproject
TRANSP:OPAQUE
UID:<deleted>
X-MICROSOFT-CDO-BUSYSTATUS:BUSY
X-MICROSOFT-CDO-IMPORTANCE:1
X-MICROSOFT-DISALLOW-COUNTER:FALSE
X-MS-OLK-AUTOFILLLOCATION:TRUE
X-MS-OLK-AUTOSTARTCHECK:FALSE
X-MS-OLK-CONFTYPE:0
END:VEVENT
END:VCALENDAR

Thanks!

**Comment by thefeiter** [comment]
Is it still possible to implement the possibility for single event colors in a future version?

**Comment by DeepDiver1975** [comment]
> Is it still possible to implement the possibility for single event colors in a future version?

as long as there is no standard which works across all clients I doubt that we can fix this ... :man_shrugging:
